src/http/content.cr

```diff
@@ -3,16 +3,50 @@ require "http/common"
 module HTTP
   # :nodoc:
   module Content
+    CONTINUE = "HTTP/1.1 100 Continue\r\n\r\n"
+
+    @continue_sent = false
+    setter expects_continue : Bool = false
+
     def close
+      @expects_continue = false
       skip_to_end
       super
     end
+
+    protected def ensure_send_continue
+      return unless @expects_continue
+      return if @continue_sent
+      @io << CONTINUE
+      @io.flush
+      @continue_sent = true
+    end
   end
 
   # :nodoc:
   class FixedLengthContent < IO::Sized
     include Content
 
+    def read(slice : Bytes)
+      ensure_send_continue
+      super
+    end
+
+    def read_byte
+      ensure_send_continue
+      super
+    end
+
+    def peek
+      ensure_send_continue
+      super
+    end
+
+    def skip(bytes_count)
+      ensure_send_continue
```

Comment:
Why do we `ensure_send_continue` in `#skip` here but not in `UnknownLengthContent`?

Reply:
Apparently because I'm just sloppy :P Sorry.